sjb/config/test_cases/ami_build_origin_int_rhel_fork.yml

```diff
@@ -0,0 +1,257 @@
+      oct prepare repositories
+      # Replace origin repo with the specified fork
+      oct sync remote origin --new-remote fork "https://github.com/${ORIGIN_FORK_ID}/${ORIGIN_REPO}"
+      oct sync remote origin --remote fork --branch "${ORIGIN_BRANCH}" --merge-into "${TARGET_ORIGIN_BRANCH}"
+  - type: "script"
+    title: "install system accounting rules"
+    script: |-
+      cat <<CONF >origin-ami-accounting.conf
+      [Manager]
+      DefaultCPUAccounting=yes
+      DefaultMemoryAccounting=yes
+      CONF
+      sudo mkdir -p /etc/systemd/system.conf.d/
+      sudo mv origin-ami-accounting.conf /etc/systemd/system.conf.d/
+      sudo systemctl daemon-reexec
+  - type: "host_script"
+    title: "package the AMI"
+    script: |-
+      oct package ami --upgrade
+  - type: "script"
+    title: "use a ramdisk for etcd"
+    script: |-
+      sudo mkdir -p /tmp
+      sudo mount -t tmpfs -o size=2048m tmpfs /tmp
+      sudo restorecon -R /tmp
+  - type: "script"
+    title: "build an origin release"
+    repository: "origin"
+    script: |-
+      export OS_BUILD_IMAGE_ARGS=''
+      hack/build-base-images.sh
+      OS_ONLY_BUILD_PLATFORMS='linux/amd64' OS_BUILD_ENV_PRESERVE="_output/local/releases:_output/local/bin" hack/env make build-rpms
+      hack/build-images.sh
+      hack/extract-release.sh
+  - type: "script"
+    title: "verify commit history"
+    repository: "origin"
+    script: |-
+      # run commitchecker outside release container as it needs
+      # access to git; also explicitly force godeps verification
+      branch="$( git rev-parse --abbrev-ref --symbolic-full-name HEAD )"
+      if [[ "${branch}" == "master" ]]; then
+        RESTORE_AND_VERIFY_GODEPS=1 make verify-commits -j
+      fi
+  - type: "script"
+    title: "run check and verify tasks"
+    repository: "origin"
+    script: |-
+      OS_BUILD_ENV_EXCLUDE='_output/local/go' hack/env TEST_KUBE='true' JUNIT_REPORT='true' make check -o build -j -k
+  - type: "script"
+    title: "run integration tests"
+    repository: "origin"
+    script: |-
+      OS_BUILD_ENV_PRESERVE=_output/local/bin/linux/amd64/integration.test hack/env make build-integration-test
+      OPENSHIFT_SKIP_BUILD='true' JUNIT_REPORT='true' make test -o check -o build -k
+  - type: "host_script"
+    title: "prepare QE test queue"
+    script: |-
+      QE_TESTS_ARRAY=()
+      if [[ "$RUN_QE_FULLFUNCTIONAL_TESTS" == "true" ]]; then
+        QE_TESTS_ARRAY+=(fullfunction)
+      elif [[ "$RUN_QE_ACCEPTANCE_TESTS" == "true" ]]; then
+        QE_TESTS_ARRAY+=(acceptance)
+      elif [[ "$RUN_QE_SMOKE_TESTS" == "true" ]]; then
+        QE_TESTS_ARRAY+=(smoke)
+      fi
+
+      if [[ ${#QE_TESTS_ARRAY[@]} -eq 0 ]]; then
+        exit 0
+      fi
+
+      QE_TESTS=""
+      for test in "${QE_TESTS_ARRAY[@]}"; do
+        if [[ -z "$QE_TESTS" ]]; then
+          QE_TESTS=$test
+        else
+          QE_TESTS=$QE_TESTS,$test
+        fi
+      done
+
+      QE_TEST_QUEUE=~/fork_ami_qe_test_queue
```

Comment:
I am not sure how the queue works, but AFAIK if developer need QE test before the PR merged, they will build a fork AMI using the PR, for the current workflow, we need a trello card to track this, if developer triggered the brew build, the  generated AMI will be pushed to AWS platform, and the robot will paste the AMI name to trello card, so QE can use this fork AMI to startup a all-in-one cluster to test this PR, So I think maybe this queues is for this purpose. there is a card here you can have a look, it a card that track a bug fix, but dev need test before it merged.
https://trello.com/c/RTqm4XYV/893-0-registry-3-3-v2-create-migration-scripts-that-will-remove-manifests-from-images-in-etcd